design/proposals/spike-cli-dockerless.md

```diff
@@ -0,0 +1,209 @@
+| Ruby CLI   | TBD                                                          |
+| Go CLI     | Pipeline maintained but the testing is not at the level it needs to be. Existing functionality will demand additional testing. <br />Considering the limited implemented functionality, new tests will need to be added if we choose to expand. |
+| Python CLI | Tested well                                                  |
+
+|            | Security |
+| ---------- | -------- |
+| Ruby CLI   | TBD      |
+| Go CLI     | TBD      |
+| Python CLI | TBD      |
+
+#### CLI commands to be supported
+
+The CLI should support the following commands:
+
+*Core* -  `init`, `authn` (`login`, `authenticate`),  `policy (load)`, `list`, `show `, `rotate_api_key` (`host` / `user`) and `fetch secrets`
+
+*Additional we might want to consider* - `resource`, `role`, `check`, `help`
+
+#### Estimation
+
+The estimations 
+
+Ruby CLI - ~3 weeks
+
+Go CLI - Complete rewrite (~10+ weeks)
+
+Python CLI - ~7 weeks
+
+#### Takeways and recommendation
+
+The following table summarizes the findings criteria detailed above and translates them to numbered equivalents. 
+
+1 - the least work
+
+3 - the most work
+
+|            | Functionality | Quality | UX   | Deployment | Testing | Security |
+| ---------- | ------------- | ------- | ---- | ---------- | ------- | -------- |
+| Ruby CLI   | 1             | 1       | 2    | 3          | 1       | TBD      |
+| Go CLI     | 3             | 3       | 3    | 2          | 3       | TBD      |
+| Python CLI | 2             | 2       | 1    | 1          | 2       | TBD      |
+
+Ruby requires the least amount of work in terms of what is already fuctionally supported but its UX is not intuitive. As part of its deployment, the CLI requires a specific version of Ruby (`2.2.5`) installed on the machine to run the CLI. Machines may not have internet access so we will need to pre-package Ruby and other dependencies and deliver as with an installer.
+
+Go CLI would require a full rewrite due to it's limited functionality, test coverage, and architectural flaws. 
+
+Python has most of the basic actions needed to interface with the Conjur server. We will need to fill in the gaps but is architecturally more stable and well tested. The Python CLI we deliver an executable so the user will not need install additional dependencies.
+
+*Recommendation*: Python CLI because of its better design and it acts as a great point of entry. In terms of a model and architectural design, it is where we want to be so it is a great first step into reforming our APIs.
+
+## Documentation
+
+[//]: # "Add notes on what should be documented in this solution. Elaborate on where this should be documented. If the change is in open-source projects, we may need to update the docs in github too. If it's in Conjur and/or DAP mention which products are affected by it"
+
+The CLI that we choose will require documentation.
```

Comment:
Add to here the fact that Ruby CLI _has_ documentation, and it will apply for go CLI if it maintains the Ruby UX.
So only Python missing documentation, or if any UX changes.